core/src/main/java/org/apache/calcite/plan/volcano/VolcanoRuleCall.java

```diff
@@ -136,7 +155,9 @@ public void transformTo(RelNode rel, Map<RelNode, RelNode> equiv,
         volcanoPlanner.ensureRegistered(
             entry.getKey(), entry.getValue());
       }
-      volcanoPlanner.ensureRegistered(rel, rels[0]);
+      RelSubset subset = volcanoPlanner.ensureRegistered(rel, rels[0]);
+      // The subset is not used, but we need it, just for debugging
+      assert subset != null;
```

Comment:
Nit:  why not use `checkArgument` or something similar from Guava? Such that it can both check expected result, stop execute when not satisfying, and provide detailed description.

Reply:
Saying this just because the comment says this assert is for debugging, so I assumed some context will be more useful for people who don't have know this much.

Reply:
It is not intended to check null or not, just to silence the IDE that the 'subset' is used. But in fact it is useful to add a breakpoint there to see the content of subset.

Reply:
Gotcha. Now I see it is to suppress IDE, style checker, etc.

Thanks for clarification.

Reply:
I find it confusing to use an assertion just to silence some warnings. Usually there are annotations for this kind of things. Apart from that if the content is useful for debugging then why not logging it for real.

Reply:
Good point, we can use annotations instead.
But logging is not what I want, I just want to add a debug point, check the returned subset for a specific rule match. Logging will print a lot of subsets for all the rule matches that I am not interested in.

Reply:
after udpate, this exception happend: java.lang.RuntimeException: rel#3291:EnumerableHashJoin.ENUMERABLE.[](left=RelSubset#88,right=EnumerableNestedLoopJoin#3290,condition==($0, $3),joinType=inner) is a PhysicalNode, which is not allowed in JoinAssociateRule